ui/src/transaction.ts

```diff
@@ -207,15 +207,25 @@ function txData(txc: ContractTransaction, date: number = Date.now()): TxData {
 function registerTxUpdateCallback(tx: Tx): void {
   if (tx.status === TxStatus.AwaitingInclusion) {
     const provider = svelteStore.get(walletStore).provider;
-    provider.waitForTransaction(tx.hash).then(txReceipt => {
-      store.update((txs: Tx[]) => {
-        const tx_ = txs.find(tx_ => tx_.hash === tx.hash)
-        if (tx_) {
-          tx_.status = status(txReceipt);
-        }
-        return txs
+    provider
+      .waitForTransaction(tx.hash)
+      .then(txReceipt => {
+        store.update((txs: Tx[]) => {
+          const tx_ = txs.find(tx_ => tx_.hash === tx.hash);
+          if (tx_) {
+            tx_.status = status(txReceipt);
+          }
+          return txs;
+        });
+      })
+      .catch(err => {
+        error.show(
+          new error.Error({
+            message: "Failed to update transaction status",
+            source: error.fromJsError(err),
```

Comment:
The following is better. It also handles cases were `err` is not a JS `Error`

```suggestion
            source: err,
```